torch/headeronly/core/ScalarType.h

```diff
@@ -267,6 +267,25 @@ enum class ScalarType : int8_t {
 constexpr uint16_t NumScalarTypes =
     static_cast<uint16_t>(ScalarType::NumOptions);
 
+inline const char* toString(ScalarType t) {
+#define DEFINE_CASE(_, name) \
+  case ScalarType::name:     \
+    return #name;
+
+  switch (t) {
+    AT_FORALL_SCALAR_TYPES_WITH_COMPLEX_AND_QINTS(DEFINE_CASE)
+    default:
+      return "UNKNOWN_SCALAR";
+  }
+#undef DEFINE_CASE
+}
+
+inline std::ostream& operator<<(
```

Comment:
We should expose what we want to use in torch::headeronly (for toString and this)

Reply:
We want `operator<<` so that ScalarType values can be used within `STD_TORCH_CHECK/TORCH_CHECK`.